expression/builtin_cast.go

```diff
@@ -170,6 +171,7 @@ type castAsRealFunctionClass struct {
 func (b *castAsRealFunctionClass) getFunction(args []Expression, ctx context.Context) (sig builtinFunc, err error) {
 	bf := baseRealBuiltinFunc{newBaseBuiltinFunc(args, ctx)}
 	bf.tp = b.tp
+	bf.tp.Flen, bf.tp.Decimal = 23, types.UnspecifiedLength
```

Comment:
why specifically set `Flen` and `Decimal` here in `castAsRealFunctionClass` while others like `castAsIntFunctionClass` do not?

`23` may be replaced with `mysql. MaxRealWidth` ?